p4-16/spec/P4-16-spec.mdk

```diff
@@ -4118,11 +4118,34 @@ h = { 10, 12 };  // This also makes the header h valid
 h = { y = 12, x = 10 };  // Same effect as above
 ~ End P4Example
 
-Two headers can be compared for equality (==) or inequality (!=) only
+Two headers can be compared for equality (`==`) or inequality (`!=`) only
 if they have the same type.  Two headers are equal if and only if they
 are both invalid, or they are both valid and all their corresponding
 fields are equal.
 
+The expression `(H) {#}` represents an invalid header of type `H`,
```

Comment:
Proposed replacement for this paragraph:

The expression `{#}` represents an invalid header.  In some contexts, the compiler may require
it to be written `(H) {#}` so that it need not infer the type of header `H` itself.

Reply:
Nate prefers that we make the same technical decision here as we do for structure-valued expressions, as to whether we call `(H)` before it a cast, vs. "some other thing that isn't a cast".